gson/src/org/immutables/gson/stream/JsonParserReader.java

```diff
@@ -143,7 +144,12 @@ public String nextString() throws IOException {
   @Override
   public boolean nextBoolean() throws IOException {
     requirePeek();
-    boolean value = parser.getBooleanValue();
+    boolean value;
+    try {
+      value = parser.getBooleanValue();
+    } catch (JsonParseException e) {
+      value = Boolean.parseBoolean(parser.getValueAsString());
```

Comment:
Is your proposed test `GsonJacksonBridgeSerializationTest`  failing without try / catch ?

Reply:
This code can be reverted. It isn't, but is not required as I have subclasses the reader. So to clarify formats that do not support tokens other than string, such as xml and property, will have to use the XmlParserReader.

Reply:
Let's keep this logic in `XmlParserReader` then. I would like to avoid making too many assumptions in `JsonParserReader` if that's OK with you.